tests/phpunit/settings_helper_test.php

```diff
@@ -0,0 +1,81 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * @package     auth_saml2
+ * @author      Tom Dickman <tomdickman@catalyst-au.net>
+ * @copyright   2019 Catalyst IT Australia {@link http://www.catalyst-au.net}
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace auth_saml2\tests;
+
+use \advanced_testcase;
+use auth_saml2\settings_helper;
+
+defined('MOODLE_INTERNAL') || die();
+
+require_once(__DIR__ . '/../../_autoload.php');
+
+/**
+ * Class settings_helper_testcase
+ *
+ * @package auth_saml2\tests
+ */
+class settings_helper_testcase extends advanced_testcase {
```

Comment:
Why does any of this file exist at all? Moodle has internal libs for parsing and validating urls, shouldn't we just use the,?

Reply:
@tomdickman  should know the exact answer as he wrote this code. It's related to saml library being very picky about URL and moodle API is not enough strict

Reply:
Replaced with PARAM_URL and removed unused code.